src/gator/core/models/timetable.py

```diff
@@ -13,131 +13,116 @@
+        """Initialize a session with the given year, season, and subsession.
 
         Args:
             year: The year of the session. Must be between 0 and 9999.
             season: The season of the session.
+            subsession: The subsession of the session.
 
         Note:
-            The year and season are passed as positional arguments for
-            convenience, but they are actually stored as keyword arguments
-            in the `kwargs` dictionary. If `year` and `season` are passed
-            as keyword arguments, they will be ignored, meaning that the
-            positional arguments always take precedence.
+            The year, season, and subsession are passed as positional arguments
+            for convenience, but they are actually stored as keyword arguments
+            in the `kwargs` dictionary. If `year`, `season`, or `subsession`
+            are passed as keyword arguments, they will be ignored, meaning that
+            the positional arguments always take precedence.
         """
         kwargs['year'] = year
         kwargs['season'] = season
+        kwargs['subsession'] = subsession
         super().__init__(*args, **kwargs)
 
     @property
     def code(self) -> str:
-        """The canonical representation of this session.
-
-        Returns:
-            A five-length string where the first four characters denote the
-            session year, and the last character denotes the session season,
-            9 for fall, 1 for winter, and 5 for summer.
+        """Return the canonical representation of the session.
 
-        Examples:
-            >>> Session(2020, 'fall').code
-            '20209'
-            >>> Session(2019, 'winter').code
-            '20191'
-            >>> Session(1966, 'summer').code
-            '19665'
-        """
-        return f'{str(self.year).zfill(4)}{self._SEASON_MAP[self.season]}'
-
-    @property
-    def human_str(self) -> str:
-        """A human-readable representation of this session.
+        The canonical representation of a session is a string whose format depends on the
+        season of the session.
 
         Examples:
-            >>> Session(2020, 'fall').human_str
-            'Fall 2020'
-            >>> Session(2019, 'winter').human_str
-            'Winter 2019'
-            >>> Session(1966, 'summer').human_str
-            'Summer 1966'
+            >>> Session(2022, 'regular', 'first').code
+            '20229'
+            >>> Session(2022, 'regular', 'second').code
+            '20231'
+            >>> Session(2022, 'regular', 'whole').code
+            '20229-20231'
+            >>> Session(2023, 'summer', 'first').code
+            '20235F'
+            >>> Session(2023, 'summer', 'second').code
+            '20235S'
+            >>> Session(2023, 'summer', 'whole').code
+            '20235'
         """
-        return f'{self.season.capitalize()} {self.year}'
+        assert self.season in ['regular', 'summer']
```

Comment:
This line is unnecessary since mongoengine actually does the validation for us (i.e. the `choices=...`) arg in the field.